jupyterlab_notify/magics.py

```diff
@@ -55,11 +56,57 @@ def _repr_mimebundle_(self, **kwargs):
 
 @magics_class
 class NotifyCellCompletionMagics(Magics):
+    smtp_class: str = Unicode(
+        "smtplib.SMTP",
+        config=True,
+        help="Fully qualified class name for the SMTP class to use",
+    )
+    smtp_args: str = Any(
+        "localhost",
+        config=True,
+        help="Arguments to pass to the SMTP class constructor, as a string",
+    )
+
     def __init__(self, shell):
         super(NotifyCellCompletionMagics, self).__init__(shell)
+        self.smpt_instace = None
+        self._create_smtp_instance()
         # Init message prompts the user for required permissions to display desktop notifications
         display(_Notification(_NotificationType.INIT))
 
+    def _create_smtp_instance(self):
+        module_name, class_name = self.smtp_class.rsplit(".", 1)
+        module = import_module(module_name)
```

Comment:
Nitpick - this could be more defensive with clearer errors when invalid values are passed